src/main/java/teammates/ui/automated/AutomatedServlet.java

```diff
@@ -39,11 +39,12 @@ public void doPost(HttpServletRequest req, HttpServletResponse resp) {
             }
 
             action.execute();
-        } catch (Exception e) {
+        } catch (Throwable t) {
             String requestUrl = req.getRequestURL().toString();
             String requestParams = HttpRequestHelper.printRequestParameters(req);
             log.severe("Exception occured while performing " + requestUrl + "|||"
-                       + requestParams + "|||" + TeammatesException.toStringWithStackTrace(e));
+                       + requestParams + "|||" + TeammatesException.toStringWithStackTrace(t));
+            throw t; // re-throw so task will be recognised as failed and GAE retry mechanism can kick in
```

Comment:
Instead of doing this, maybe setting the response status to something not 200-299 will be better. ([cron job reference](https://cloud.google.com/appengine/docs/standard/java/config/cron) / [task queue reference](https://cloud.google.com/appengine/docs/standard/java/taskqueue/push/)).

Reply:
I think we can continue to use 100 for exceptions deliberately caught by us, but 500 for unexpected errors caught only by the final `Throwable` catch.

Reply:
Isn't 100 [already defined](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml)?

Reply:
@LiHaoTan Perhaps, but it doesn't really matter since these workers are not user-facing. Anyway, if we decide to discuss it we should do it in another issue. This is about `AutomatedServlet`.